includes/embeds/class-amp-core-block-handler.php

```diff
@@ -339,6 +339,8 @@ public function ampify_navigation_block( $block_content, $block ) {
 			add_action( 'wp_print_footer_scripts', [ $this, 'dequeue_block_navigation_view_script' ], 0 );
 		}
 
+		$is_interactive_block = false !== strpos( $block_content, 'data-wp-interactive' );
```

Comment:
This could be made more robust. What if someone uses `data-wp-interactive` in the text of the block?

I guess that's true for the other existing code below.

At the very least:

```suggestion
		$is_interactive_block = false !== strpos( $block_content, ' data-wp-interactive' );
```

Reply:
What about checking this attribute in the `<nav>` attributes? Something similar to:

```php
$is_interactive_block = str_contains(
	preg_match( '/(?<=<nav)\s[^>]+/', $block_content, $matches ) ? $matches[0] : '',
	' data-wp-interactive',
);
```

Surprisingly adding space before ` data-wp-interactive` is failing here, because it's added as a newline in the outputted block content HTML.

Reply:
Also, since it's a navigation block, should we expect such a string in block text?

Reply:
Yeah, that `preg_match()` seems like it would do the trick.

> Also, since it's a navigation block, should we expect such a string in block text?

Maybe if someone likes to blog about different attributes and has a link to a page all about this one 😄